test/smoke.go

```diff
@@ -0,0 +1,304 @@
+package test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"go/build"
+	"math/rand"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"reflect"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/snyk/kubernetes-scanner/internal/backend"
+	"github.com/snyk/kubernetes-scanner/internal/config"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	k8sconfig "sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+const scannerNamespace = "kubernetes-scanner"
+
+// Smoke runs the scanner's smoke test; deploying the scanner into a Tilt environment and ensuring
+// that the backend has received the resources.
+// The Smoke tests has a couple of prerequisites:
+// - the kubectl context needs to point to a running cluster (where tilt won't block on)
+// - tilt, kubectl and helm need to be installed.
+func Smoke(ctx context.Context) (err error) {
```

Comment:
I'll move it back to the `smoke_test.go` file, as it doesn't need to be imported in the store (as discussed in Slack).

However I'd argue that `func Smoke` is fine because it's used as `test.Smoke(...)`, while `test.SmokeTest` stutters :)

Reply:
Maybe to clarify, I've had this in a non-test file so that it can be imported from the store (sorry about that confusion!), so that the store could have had a `smoke_test.go` file:
```go
import (
    ...
    "github.com/snyk/kubernetes-scanner/test"
)

func TestSmoke(t *testing.T) {
    if err := test.Smoke(ctx); err != nil {
        t.Fatalf("%v", err)
    }
}
```

Using `testing.T` in a non-test file is possible, although might be a bit awkward? (I don't have a strong opinion on this)

---

Either way, none of that is necessary now that we just invoke the scanner's tests directly in the store as well😄

Reply:
True